### DISCUSSION TITLE
New to GitHub; Using brew on M1 macOS; very simple gs usage problem; how to display a postscript file?

### DISCUSSION THREAD

**Discussion by kwolcott** [Homebrew/discussions#3508] [Everyday usage]
New to GitHub; Using brew on M1 macOS; very simple gs usage problem; how to display a postscript file?

**Comment by carlocab** [comment]
You'll need to consult the gs documentation for this. Try https://www.ghostscript.com/.

**Reply by kwolcott** [reply]
I used to use a Solaris machine a very long time ago and then more recently a series of Linux machines.  I could just do type 'gs filename.ps' and it would just display.  I guess I was using XWindows in both cases; on the Mac, I'm not.

When I do 'gs filename.ps' from the terminal prompt, I get a GS> prompt.

When I do 'gs -h' I get some useful information about supported devices, it looks like all of them are printers.

When I look at the documentation provided by brew regarding gs, it doesn't really help with this problem as far as I can tell.

I think that the problem is that there is no XWindows running.

I've installed xquartz, but that doesn't seem to help.

I've been looking at the documentation you pointed me to.

It looks like there's no obvious support for viewing postscript files, just printing them.

Thanks,
Ken

**Reply by alebcay** [reply]
As it currently stands, Homebrew's copy of Ghostscript does not link against any window manager (e.g. XQuartz). When I run `gs <some-filename>`, I see command output like:
```
%%BoundingBox: 14 14 275 815
%%HiResBoundingBox: 14.166000 14.166000 274.175992 814.175975
>>showpage, press <return> to continue<<
```
When I press Enter, it returns to the `GS>` prompt as you describe. I've never viewed PS files with `gs` but I guess the expectation is that while it's paused on `showpage`, there should've been some kind of window popup.

In a pinch, installing Ghostscript also provides the `ps2pdf` command which can allow for display in Apple's native Preview application.

**Comment by kwolcott** [comment]
Thank you all for your responses.

I think that this can be closed now.

I see that I'm not going to be able to view postscript files with the gs
that can be installed from homebrew even if I have an X WIndows environment
from xquartz as gs (from homebrew) does not have X Windows support built in.

Thanks,
Ken Wolcott
